src/lib/core/TLVReader.cpp

```diff
@@ -583,23 +583,30 @@ CHIP_ERROR TLVReader::Next()
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR TLVReader::Expect(Tag expectedTag)
```

Comment:
Please add specific unit tests for those new methods in the TLV Reader unit tests.

Reply:
https://github.com/project-chip/connectedhomeip/pull/30188